libs/libc/stdio/lib_fopen.c

```diff
@@ -209,42 +206,39 @@ int lib_mode2oflags(FAR const char *mode)
                   {
                     /* Retain any binary and exclusive mode selections */
 
-                    oflags &= (O_BINARY | O_EXCL);
+                    oflags &= FLAG_KEEP;
```

Comment:
I think since `O_TEXT` is non-zero now then we need to update other parts in `lib_mode2oflags` like
```
          /* Open for read access ("r{b|x|+}") */

          case 'r' :
            if (state == MODE_NONE)
              {
                /* Open for read access */

                oflags = O_TEXT | O_RDOK;
                state  = MODE_R;
              }
            else
              {
                goto errout;
              }
            break;

          /* Open for write access ("w{b|x|+}") */

          case 'w' :
            if (state == MODE_NONE)
              {
                /* Open for write access, truncating any existing file */

                oflags = O_TEXT | O_WROK | O_CREAT | O_TRUNC;
                state  = MODE_W;
              }
            else
              {
                goto errout;
              }
            break;

          /* Open for write/append access ("a{b|x|+}") */

          case 'a' :
            if (state == MODE_NONE)
              {
                /* Write to the end of the file */

                oflags = O_TEXT | O_WROK | O_CREAT | O_APPEND;
                state  = MODE_A;
              }
            else
              {
                goto errout;
              }
            break;
```

Reply:
All these place is unified in one macro(FLAG_KEEP):
https://github.com/apache/incubator-nuttx/pull/6357/commits/57567c310c8d6b7087434bea38bc39df1d75dad9

Reply:
I do not see how `FLAG_KEEP` address issue that I'm writing about. Could you please help me to trace the flow?

Reply:
We can add `int oflags = O_TEXT;` at line 137 I think. Then all other logic can be kept without changes

Reply:
Do you want that the default behavior change to the text mode? My intention is the opposite: the default is the binary mode, that's why I change O_BINARY to zero and O_TEXT to non zero, because:

1. The text mode will change the content(\r\n->\n) on the fly, so it's better to enable this feature explicitly
2. POSIX(*BSD, Linux and NuttX) doesn't support the conversion of \r\n->\n at all, which mean the binary mode is the only( and default) mode

Reply:
For C-API I think text is the default, isn't it?

Reply:
There is "b" mode for binary, so there is a way to switch from default text, but not the opposite.

Reply:
Also I think that current O_BINARY non-zero is done exactly to have a straightforward translation from stdio API. But mapping O_BINARY seems to give more advantages

Reply:
It look like fopen default to the text mode from:
https://stackoverflow.com/questions/2174889/whats-the-differences-between-r-and-rb-in-fopen
So, I follow your suggestion that:

1. fopen is in the text mode by default
2. Need append "b" to get the binary mode
3. open is in the binary mode by default
4. Need or O_TEXT to get the text mode

Please review agian.

Reply:
I still think that line 137 change can give the same result, but I'm fine with the current change

Reply:
> I still think that line 137 change can give the same result, but I'm fine with the current change

No, the assignment at line 159, 175 and 191 will overwrite to the binary mode.

Reply:
Yes. Change `=` to `|=` is needed indeed with that

Reply:
> There is "b" mode for binary, so there is a way to switch from default text, but not the opposite.

could use "t" to switch back the text mode(https://github.com/apache/incubator-nuttx/pull/6357/files#diff-7b1af1fbf5a952ed1f0e0384ccf15b82ea5b3a0072ec125c83aeb6fb2a983795R298-R309):
https://www.ibm.com/docs/en/zos/2.4.0?topic=functions-fopen-open-file
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/fopen-wfopen?view=msvc-170

Reply:
Yes. Seems to be some extension because I can't find "t" in https://pubs.opengroup.org/onlinepubs/9699919799/functions/fopen.html